src/Umbraco.Web/Install/Controllers/InstallApiController.cs

```diff
@@ -216,11 +218,11 @@ internal bool StepRequiresExecution(InstallSetupStep step, JToken instruction)
         }
 
         // executes the step
-        internal async Task<InstallSetupResult> ExecuteStepAsync(InstallSetupStep step, JToken instruction)
+        internal async Task<InstallSetupResult> ExecuteStepAsync(InstallSetupStep step, object instruction)
         {
             using (_proflog.TraceDuration<InstallApiController>($"Executing installation step: '{step.Name}'.", "Step completed"))
             {
-                var model = instruction?.ToObject(step.StepType);
+                var model = Convert.ChangeType(instruction, step.StepType);
```

Comment:
Does this always work? Our extension method TryConvertTo is a superior conversion tool than .net's built in ones like this